pkg/providers/tinkerbell/comparator.go

```diff
@@ -0,0 +1,51 @@
+package tinkerbell
+
+import (
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1"
+)
+
+// VersionComparable enables kubernetes version retrieval.
+type VersionComparable interface {
+	Version() string
+}
+
+// MachineReplicaComparable enables replica number retrieval.
+type MachineReplicaComparable interface {
+	Replicas() int
+}
+
+// KubeadmControlPlaneComparator wraps the KubeadmControlPlane to enable Comparator functions.
+type KubeadmControlPlaneComparator struct {
```

Comment:
Similar to other comments, I don't think `Comparator` is adding any value. Its also a misrepresentation because this object doesn't provide any comparison behavior.

Reply:
Was having a difficult time coming up with an appropriate name for this. Updated this to `KubeadmControlPlaneClusterInfoRetriever` does that work?

Reply:
renaming to `KubeadmControlPlane`